src/BaGet.Hosting/FileProvider/BaGetUIFileProvider.cs

```diff
@@ -0,0 +1,136 @@
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.FileProviders.Physical;
+using Microsoft.Extensions.FileSystemGlobbing;
+using Microsoft.Extensions.Primitives;
+
+namespace BaGet.Hosting
+{
+    public class BaGetUIFileProvider : IFileProvider
+    {
+        private readonly DirectoryInfo _root;
+        private readonly string _pathBase;
+        private readonly PhysicalFilesWatcher _watcher;
+        private readonly Matcher _matcher;
+        private readonly ConcurrentDictionary<string, CachedFileInfo> _fileCache;
+
+        public BaGetUIFileProvider(DirectoryInfo root, string pathBase)
+        {
+            _root = root;
+            _pathBase = pathBase;
+            _watcher = new PhysicalFilesWatcher(root.FullName, new FileSystemWatcher(root.FullName), true);
+            _matcher = new Matcher().AddInclude("**/index.html").AddInclude("**/*.js");
```

Comment:
Do we need to match the `index.html` file? I believe only `.js` files have placeholders today. We might be able to replace this with a `Path.GetExtensions(subpath) == ".js"`

Reply:
You're absolutely right.